### PR TITLE
mode selection: further optimized packer with mode selection

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

#### Description

This PR is to make the packer to better perform with mode seletion feature. I need to test that all SymbiFlow xc7 pass.

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: vpr/src/pack/cluster.cpp [hunk 1]
```diff
@@ -1169,15 +1175,43 @@ static enum e_block_pack_status try_pack_molecule(
 				}
 			}
 			if (block_pack_status == BLK_PASSED) {
-				/* Try to route if heuristic is to route for every atom
-					Skip routing if heuristic is to route at the end of packing complex block
-				*/
-				bool is_mode_conflict = true;
+				/*
+				 * during the clustering step of `do_clustering`, `detailed_routing_stage` is incremented at each iteration until it a cluster
```
Comment (litghost) [inline vpr/src/pack/cluster.cpp]: Fix this formatting to conform to an 80 character maximum.  I recommend pulling comment out of here and putting it somewhere else.

File: vpr/src/pack/cluster.cpp [hunk 2]
```diff
@@ -2008,92 +2044,92 @@ void add_cluster_molecule_candidates_by_connectivity_and_timing(t_pb* cur_pb,
...
                     bool success = true;
                     for (int j = 0; j < get_array_size_of_molecule(molecule); j++) {
-							if (molecule->atom_block_ids[j]) {
-								VTR_ASSERT(atom_ctx.lookup.atom_clb(molecule->atom_block_ids[j]) == ClusterBlockId::INVALID());
-								auto blk_id2 = molecule->atom_block_ids[j];
-								if (!exists_free_primitive_for_atom_block(cluster_placement_stats_ptr, blk_id2)) {
-                                    /* TODO: debating whether to check if placement exists for molecule (more
-                                     * robust) or individual atom blocks (faster) */
-									success = false;
-									break;
```
Comment (litghost) [inline vpr/src/pack/cluster.cpp]: Can you remove changes that are purely white space changes?

File: vpr/src/pack/cluster.cpp (hunk 1 above)
Comment (acomodi) [reply vpr/src/pack/cluster.cpp]: Done

File: vpr/src/pack/cluster.cpp (hunk 2 above)
Comment (acomodi) [reply vpr/src/pack/cluster.cpp]: Done

**Comment by acomodi** [comment]
This was already included in upstream. Closing

**Pull request closed without merging**